pkg/controller/jobframework/reconciler.go

```diff
@@ -136,11 +137,19 @@ func NewReconciler(
 	}
 }
 
-func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Request, job GenericJob) (ctrl.Result, error) {
-	object := job.Object()
+func (r *JobReconciler) ReconcileGenericJobWrapper(ctx context.Context, req ctrl.Request, job GenericJob) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx).WithValues("job", req.String(), "gvk", job.GVK())
 	ctx = ctrl.LoggerInto(ctx, log)
 
+	result, err := r.ReconcileGenericJob(ctx, req, job)
+
+	return result, r.ignoreUnretryableError(ctx, err)
+}
+
+func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Request, job GenericJob) (ctrl.Result, error) {
```

Comment:
Does this need to be exported?

Otherwise, the names could be: `ReconcileGenericJob` and `reconcileGenericJobInternal` and add a comment that the internal function doesn't do error handling.

Or, you can just keep the original function, add names to the return parameters and add:

```golang
defer func() {
  err = r.ignoreUnretryableError(ctx, err)
}
```